UriGenerator.php

```diff
@@ -106,4 +123,16 @@ public function resolveConflict(UriContext $uriContext)
 
         return $uri;
     }
+
+    /**
+     * Configure options which apply to each token provider
+     *
+     * @param OptionsResolver
+     */
+    private function configureGlobalOptions(OptionsResolver $optionsResolver)
+    {
+        $optionsResolver->setDefaults(array(
+            'collapse_empty' => false,
```

Comment:
Well, `collapse_empty` hints that it does more than allowing empty, and it also allows to add a real `allow_empty` in the future if required.